spec.md

```diff
@@ -179,17 +179,163 @@ physically present in the final circuit. Refer to the description of the
+lowering convention.  One such strategy is `"bind"`{.firrtl} which lowers to
+modules and instances which are instantiated using the SystemVerilog
+`bind`{.verilog} feature.
+
+The `group`{.firrtl} keyword defines an optional group.  An optional group may
+only be defined inside a module.  An optional group must reference a group
+declared in the current circuit.  Statements in an optional groups may only read
+from ports or declarations of the current module or the current
+group---statements in a group may not drive components declared outside the
+group.  Declarations of identifiers in a group may not collide with identifiers
+declared outside the group.
+
+The circuit below contains one optional group declaration, `Bar`.  Module `Foo`
+contains a group definition that creates a node computed from a port defined in
+the scope of `Foo`.
+
+``` firrtl
+circuit Foo:
+  declgroup "bind", Bar:  ; Declaration of group Bar
+
+  module Foo:
+    input a: UInt<1>
+
+    group Bar:            ; Definition of group Bar inside module Foo
+      node notA = not(a)
+
+  extmodule Baz:
+    input a: UInt<1>
+    output b: UInt<1>
+    declgroup "bind" Bar: ; Declaration of group Bar for extmodule Baz
+```
+
+Optional group declarations may be nested.  Optional group declarations are
+declared with the `declgroup`{.firrtl} keyword indented under an existing
+`declgroup`{.firrtl} keyword.  The circuit below contains four optional group
+declarations, three of which are nested.  `Bar` is the top-level group.  `Baz`
+and `Qux` are nested under `Bar`.  `Quz` is nested under `Qux`.
+
+``` firrtl
+circuit Foo:
+  declgroup "bind", Bar:
+    declgroup "bind", Baz:
+    declgroup "bind", Qux:
+      declgroup "bind", Quz:
+```
+
+Optional group definitions must match the nesting of declared groups.  Optional
+groups are defined under existing groups with the `group`{.firrtl} keyword
+indented under an existing `group`{.firrtl} keyword.  For the four declared
+groups in the circuit above, the following is a legal nesting of group
+definitions:
+
+``` firrtl
+module Foo:
+  input a: UInt<1>
+
+  group Bar:
+    node notA = not(a)
+    group Baz:
+    group Qux:
+      group Quz:
+        node notNotA = not(notA)
+```
+
+Statements in a nested optional group may only read from ports or declarations
+of the current module, the current group, or a parent group---statements in a
+group may not drive components declared outside the group.  In the above
+example, `notA` is accessible in the group definition of `Quz` because `notA` is
+declared in a parent group.
+
+Group instances may declare output, passive ports of `Probe`{.firrtl} type.
+These ports are accessible in parent modules which instantiate the child.  In
+the example below, module `Baz` defines a group `Bar` with an output port `_a`.
+This port is accessible inside module `Foo` within its defined group `Bar`.
+_Put differently, module `Baz` has an additional port `_a` that is only
```

Comment:
Stray "_".

Reply:
This is making the sentence italicized. It is terminated at the end of the sentence. 😉